### ISSUE TITLE
Performance hit between 0.8.5 and 0.8.6

### ISSUE THREAD

**Issue by RuthAdele** [dompdf/dompdf#2288]
I recently upgraded from 0.8.5 to 0.8.6 in order to fix an issue with the PDF not rendering .png images.
While this "fixed" the png rendering issue, it's caused massive performance issues with how long it took for the PDF to generate.

Running on PHP 7.3

Went from 1-2 secs load to 20-30 secs load time.

All server configuration as per the documentation (i.e. dom, mbstring, php-font-lib, php-svg-lib, opcache, gd, imagick extensions all enabled).

**Comment by bsweeney** [comment]
Can't really say why you're seeing a performance decrease without seeing a sample of the HTML + CSS.

**Comment by halfpastfouram** [comment]
@RuthAdele can you try and profile with xdebug? That way can see exactly what part of the code is responsible for the increase in load time.

**Comment by aex1701** [comment]
Same here. In my case there is a massive amount of empty requests during the pdf generation with 0.8.6. 0.8.5 works without problems.
![image](https://user-images.githubusercontent.com/2996397/98711137-e6d05400-2384-11eb-8d74-51b6a06f125e.png)

**Comment by bsweeney** [comment]
@aex1701 using drupal? The "/none" url looks like what has been reported in #2244.

**Comment by aex1701** [comment]
No, i am using the Laravel framework with the Laravel DomPDF Wrapper https://github.com/barryvdh/laravel-dompdf

In my case there seems to be a problem with the base Tag. Without base Tag there are no "none" calls.

With the following HTML Code i have exactly one "none" call:
![image](https://user-images.githubusercontent.com/2996397/98840505-2d868280-2447-11eb-8ee5-27f7e7e48ebc.png)

**Comment by bsweeney** [comment]
FYI sample code is more helpful as text so its copyable. And for debugging Dompdf the preference is to see the actual HTML + CSS.

However, this is helpful. So far I am unable to confirm, but what it sounds like is that the base href is being appended to the background image keyword "none" (the default) making it no longer the keyword but "/none". Can you provide any additional relevant details such as the actual HTML as passed to Dompdf and any settings you're using?

**Comment by bsweeney** [comment]
I've confirmed there is an issue when a base href is provided. There's a bug in the CSS image parsing logic. I'm not sure why the image cache doesn't mitigate excessive calls, so that's also something to look into.

@RuthAdele are you able to confirm this on your installation. The excessive calls to an invalid URL would definitely cause a performance hit.

**Comment by PowerKiKi** [comment]
I can confirm the bug, first seen in a TYPO3 environnement, but then reproduced completely independently, as shown below.

In order to reproduce you must have access to a server where you can see logs. It doesn't matter whether it is a local server or remote one, as long as you can see the logs. And I assume the server reply does not matter either, but in my case it was answering 404 (but containing generated HTML which slowed down things even more).

```php
<?php

use Dompdf\Dompdf;
use Dompdf\Options;

require 'vendor/autoload.php';

$html = '
<html>
    <head>
        <!-- Here use an absolute URL to a server where you can see logs -->
        <base href="https://my-server.com/">
    </head>
    <body>
        Each span will trigger HTTP fetch, so add more span to make it slower
        <span>foo</span>
        <span>foo</span>
        <span>foo</span>
    </body>
</html>
';

$options = new Options();
$options->setIsRemoteEnabled(true); // This is required to see the bug

// Allow self signed certificate for easier local development
$context = stream_context_create([
    'ssl' => [
        'allow_self_signed' => true,
    ],
]);

$dompdf = new Dompdf($options);
$dompdf->setHttpContext($context);
$dompdf->loadHtml($html);

// This will be super slow if your server happen to be not extremely fast to answer tens, hundreds or thousands (!) of requests to /none
$dompdf->render();

// This is not required but it's nice to see the PDF is generated correctly, though extremely slowly
$dompdf->stream('test.pdf', ['Attachment' => false]);
```

In my case, my `/var/log/nginx/access.log` will show:

```
127.0.0.1 - - [10/Dec/2020:11:47:02 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
127.0.0.1 - - [10/Dec/2020:11:47:04 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
127.0.0.1 - - [10/Dec/2020:11:47:07 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
127.0.0.1 - - [10/Dec/2020:11:47:09 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
127.0.0.1 - - [10/Dec/2020:11:47:12 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
127.0.0.1 - - [10/Dec/2020:11:47:14 +0900] "GET https://my-server.com/none HTTP/1.0" 307 0 "-" "-" "-"
[...]
```

This was with dompdf v0.8.6. And I can confirm that requiring dev-develop on 818fa5b will fix the issue.

As this seems a rather important regression, is there any timeframe for the next release that would fix this ? Maybe as a 0.8.7 if 1.0.0 is not ready yet ?

**Comment by bsweeney** [comment]
Thanks for the feedback. Now that we're behind the PHP 8 release by a few weeks I want to clear out the 1.0.0 milestone, moving most of the issues to the 1.1.0 milestone for now, and release what we have. Hopefully in the next few days 🤞
